machine_manager_preventive/models/preventive_machine_operation.py

```diff
@@ -10,65 +10,91 @@ class PreventiveMachineOperation(models.Model):
     _name = "preventive.machine.operation"
     _description = "Preventive operation for machine."
 
-    @api.one
+    @api.multi
+    @api.depends('cycles')
     def _next_cycles(self):
-        if self and self.cycles:
-            self.nextcycles = self.lastcycles + self.cycles
-
-    name = fields.Char('REF', required=True)
-    opdescription = fields.Text('Description')
-    machine = fields.Many2one('machinery', 'Machine', required=True,
-                              readonly=True)
-    opname_omm = fields.Many2one('preventive.operation.matmach',
-                                 'Operation Material Machine', required=True,
-                                 readonly=True)
-    frequency = fields.Integer('Frequency',
+        for record in self:
+            if record and record.cycles:
+                record.nextcycles = record.lastcycles + record.cycles
+
+    @api.multi
+    @api.depends('lastdate', 'frequency', 'interval_unit')
+    def _next_date(self):
+        for record in self:
```

Comment:
si haces `self.filtered(lambda x: x.lastdate and x.frequency and x.interval_unit) `y te ahorras el if